networkx/convert_matrix.py

```diff
@@ -425,6 +427,203 @@ def from_pandas_edgelist(
+        >>> gdf = gpd.GeoDataFrame(zip(names, lines), columns=["name", "geometry"], geometry="geometry")
+        >>> G = nx.from_geopandas_edgelist(gdf, "geometry", edge_attr=list(gdf.columns))
+        >>> G[0][1]["name"]
+        'A'
+        >>> G[1][2]["geometry"].length
+        2.8284271247461903
+
+        """
+
+    try:
+        from shapely.geometry import Point
+    except ImportError:
+        # A bit unintuitive - we need to import shapely but it is a core dependency of geopandas, the library of specific interest to the user
+        raise ImportError(
+            "Creating a graph from a GeoDataFrame requires geopandas to be installed."
+        )
+
+    # If the geodataframe is empty, just return an empty graph/digraph
+    if gdf.empty:
+        return nx.empty_graph(0, create_using)
+
+    # Determine number of dimensions of geometry by checking the first row (i.e. 2D or 3D lines?)
+    dims = range(len(gdf[geometry][0].coords[0]))
+
+    # Find all unique start & end points and assign them an id
+    gdf["nx_source_coords"] = gdf[geometry].apply(
+        # lambda i: (round(i.coords[0][0], precision), round(i.coords[0][1], precision), round(i.coords[0][2], precision))
+        lambda geom: tuple(round(geom.coords[0][i], precision) for i in dims)
+    )
+    gdf["nx_target_coords"] = gdf[geometry].apply(
+        # lambda i: (round(i.coords[-1][0], precision), round(i.coords[-1][1], precision), round(i.coords[-1][2], precision))
+        lambda geom: tuple(round(geom.coords[-1][i], precision) for i in dims)
+    )
+    node_ids = {}
+    i = 0
+    for row in gdf.itertuples(index=False):
+        node_1 = row.nx_source_coords
+        node_2 = row.nx_target_coords
+        if node_1 not in node_ids:
+            node_ids[node_1] = i
+            i += 1
+        if node_2 not in node_ids:
+            node_ids[node_2] = i
+            i += 1
+
+    # Assign the unique id to each
+    gdf["nx_source"] = gdf["nx_source_coords"].apply(lambda x: node_ids[x])
+    gdf["nx_target"] = gdf["nx_target_coords"].apply(lambda x: node_ids[x])
+
+    # Make the graph
+    if edge_attr is True:
+        edge_attr = [attr for attr in gdf.columns if attr[:3] != "nx_"]
+    graph = nx.from_pandas_edgelist(
+        gdf,
+        source="nx_source",
+        target="nx_target",
+        edge_attr=edge_attr,
+        create_using=create_using,
+        edge_key=edge_key,
+    )
+
+    # Assign new point geometries to the nodes
+    node_geoms = {node_ids[k]: Point(k) for k in node_ids}
+    for n in node_geoms:
+        graph.nodes[n].update(geometry=node_geoms[n])
+
+    # Drop these intermediate columns
+    gdf.drop(
+        ["nx_source_coords", "nx_target_coords", "nx_source", "nx_target"],
+        axis="columns",
+        inplace=True,
+    )
+
+    return graph
+
+
+def to_geopandas_edgelist(
+    G, source="source", target="target", geometry="geometry", crs=None, nodelist=None,
```

Comment:
```geometry``` and ```crs``` should be added to the doc_string of ```to_geopandas_edgelist```.
Perhaps you should change the order of the parameters to list ```nodelist``` before ```geometry``` or ```crs```?

Reply:
Also, the ```nodelist``` description should be tweaked a little. The edges that are added have starting node in ```nodelist```. It is not true that edges must include both incident nodes in ```nodelist```. 

Perhaps "Use only nodes specified in nodelist" -> "Include only edges that start at one of the nodes in nodelist"